packages/flutter_tools/test/general.shard/compute_dev_dependencies_test.dart

```diff
@@ -30,393 +29,110 @@ const String _dartBin = 'bin/cache/dart-sdk/bin/dart';
 //     pubspec.yaml
 void main() {
   late FileSystem fileSystem;
-  late FlutterProject project;
-  late BufferLogger logger;
 
   setUp(() {
     Cache.flutterRoot = '';
     fileSystem = MemoryFileSystem.test();
-    project = FlutterProject.fromDirectoryTest(fileSystem.currentDirectory);
-    logger = BufferLogger.test();
   });
 
-  Pub pub(ProcessManager processManager) {
-    return Pub.test(
-      fileSystem: fileSystem,
-      logger: logger,
-      processManager: processManager,
-      platform: FakePlatform(),
-      botDetector: const FakeBotDetector(false),
-      stdio: FakeStdio(),
-    );
+  void writePackages(List<Package> graph) {
+    final Map<String, Object?> packageConfigMap = <String, Object?>{'configVersion': 2};
+    for (final Package package in graph) {
+      fileSystem.file(fileSystem.path.join(package.name, 'pubspec.yaml'))
+        ..createSync(recursive: true)
+        ..writeAsStringSync('''
+name: ${package.name}
+dependencies:
+${package.dependencies.map((String d) => '  $d: {path: ../$d}').join('\n')}
+dev_dependencies:
+${package.devDependencies.map((String d) => '  $d: {path: ../$d}').join('\n')}
+''');
+      ((packageConfigMap['packages'] ??= <Object?>[]) as List<Object?>).add(<String, Object?>{
+        'name': package.name,
+        'rootUri': '../../${package.name}',
+        'packageUri': 'lib/',
+        'languageVersion': '3.7',
+      });
+    }
+    fileSystem.file(fileSystem.path.join(graph.first.name, '.dart_tool', 'package_config.json'))
+      ..createSync(recursive: true)
+      ..writeAsStringSync(jsonEncode(packageConfigMap));
   }
 
-  test('no dev dependencies at all', () async {
-    // Simulates the following:
-    //
-    // # /my_app/pubspec.yaml
-    // name: my_app
-    // dependencies:
-    //   package_a:
-    //
-    // # /package_a/pubspec.yaml
-    // name: package_a
-    // dependencies:
-    //   package_b:
-    final ProcessManager processes = _dartPubDepsReturns('''
-    {
-      "root": "my_app",
-      "packages": [
-        {
-          "name": "my_app",
-          "kind": "root",
-          "dependencies": [
-            "package_a",
-            "package_b"
-          ],
-          "directDependencies": [
-            "package_a"
-          ],
-          "devDependencies": []
-        },
-        {
-          "name": "package_a",
-          "kind": "direct",
-          "dependencies": [
-            "package_b"
-          ],
-          "directDependencies": [
-            "package_b"
-          ]
-        },
-        {
-          "name": "package_b",
-          "kind": "transitive",
-          "dependencies": [],
-          "directDependencies": []
-        }
-      ]
-    }''', project: project);
+  Future<void> validateComputeGraph(
```

Comment:
Ditto, what is it validating? Can you explain in the comment, thanks!

Reply:
It should be renamed - I believe at one point I called `computeTransitiveDependencies` `computeTransitiveDependencies`... 
I'll add a comment also